fuels-contract/src/contract.rs

```diff
@@ -82,16 +82,23 @@ impl Contract {
         wallet: LocalWallet,
     ) -> Result<Vec<Receipt>, Error> {
         // Script to call the contract. The offset that points to the `script_data` is loaded at the
-        // register `0x12`. Note that we're picking `0x12` simply because it could be any
+        // register `0x13`. Note that we're picking `0x13` simply because it could be any
         // non-reserved register. Then, we use the Opcode to call a contract: `CALL` pointing at the
         // register that we loaded the `script_data` at.
+        let forward_data_offset = ContractId::LEN + WORD_SIZE;
         let (script, offset) = script_with_data_offset!(
             data_offset,
             vec![
                 // Load call data to 0x10.
-                Opcode::MOVI(0x10, data_offset + 32),
+                Opcode::MOVI(0x10, data_offset + forward_data_offset as Immediate18),
```

Comment:
Why is it cast as an Immediate18 if we want to support `u64` (which would be a word-size, not a u32 like Immediate18). I'm not familiar with this.

Reply:
That's an address, and it's the address of the call data. The subsequent two instructions move the memory address of the 8-byte amount in script data into a register, than load from that memory address. That's what does the `u64` amount handling.

Reply:
Oh ok!!! ty for the explaination